moshi/src/test/java/com/squareup/moshi/JsonWriterFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi;
+
+import java.util.Arrays;
+import java.util.List;
+import okio.Buffer;
+
+abstract class JsonWriterFactory {
```

Comment:
This one’s stateful and so I changed JsonReaderFactory to be stateful for symmetry.